code/script.js

```diff
@@ -26,22 +42,221 @@ const showMessage = (message, sender) => {
+  const userName = inputValue.value
+
+  // Check whether input box is empty, interrupt chat if that's the case 
+
+  if (userName){
+    showMessage(`My name is ${userName}`, 'user')
+    setTimeout(moodQuestion, 1000) // Move to next bot question
+  } else {
+    showMessage(`Whoops 🙈 Please refresh the page and fill in your name!`, 'bot')
+  } 
+  inputValue.value = ''
+
+}
+
+const handleMoodQuestion = () => {
+  const moodOption = inputValue.value
+
+  showMessage(`I\m feeling ${moodOption}`, 'user')
+
+  inputValue.value = ''
+
+  setTimeout(followUpQuestion, 1000) 
+
+}
+
+// Propose different activities depending on the user's mood
+
+const handleSadActivity = (userAnswer) => {
+
+  userAnswer = document.getElementById('select').value
+  // Store URL link in a variable to use in conditionals
+  const supportHotline = `<a href="https://mind.se/hitta-hjalp/" target="_blank">Mind.se</a>`;
+  showMessage(userAnswer, 'user')
+
+  if(userAnswer === 'yes'){
+    showMessage(`This organization has specialized support hotlines you can contact. Give it a try:`, 'bot')
+    showMessage(supportHotline, 'bot')
+  } else {
+    showMessage(`OK, I understand. I\'m here if you need me! Take care.`, 'bot')
+  }
+}
+
+const handleNeutralActivity = (userAnswer) => {
+  
+  userAnswer = document.getElementById('select').value
+  const cuteAnimals = `<a href="https://www.instagram.com/explore/tags/cuteanimals/" target="_blank">Instagram's cute animals</a>`;
+  showMessage(userAnswer, 'user')
+
+  if(userAnswer === 'yes'){
+    showMessage(`Check out Instagram's "cute animals" hashtag! But make sure to set a timer, because it's quite addictive 😊`, 'bot')
+    showMessage(cuteAnimals, 'bot')
+  } else {
+    showMessage('Gotcha! I\'m here if you change your mind 😊', 'bot')
+  }
+}
+
+const handleWorriedActivity = () => {
+
+  userAnswer = document.getElementById('select').value
+  const meditation = `<a href="https://www.youtube.com/watch?v=kALrH8K7rDI" target="_blank">10min. meditation 🧘</a>`;
+  showMessage(userAnswer, 'user')
+
+  if(userAnswer === 'yes'){
+    showMessage(`Great! I recommend a YouTube channel called "Guided Meditation". Check it out: `, 'bot')
+    showMessage(meditation, 'bot')
+  } else {
+    showMessage(`OK, I understand. I\'m here if you need me. Sending you a big hug!`, 'bot')
+  }
+}
+
+const handleHappyActivity = () => {
+
+  userAnswer = document.getElementById('select').value
+  const happySong = `<a href="https://www.youtube.com/watch?v=GbpnAGajyMc" target="_blank">Come on Eileen</a>`;
+  showMessage(userAnswer, 'user')
+
+  if(userAnswer === 'yes'){
+    showMessage(`Awesome! This is one of my favorite songs: `, 'bot')
+    showMessage(happySong, 'bot')
+  } else {
+    showMessage('Alright! I\'m here if you change your mind 😊 Enjoy your day!', 'bot')
+  }
+}
+
+// Event Listeners
+
+startButton.addEventListener('click', () => setTimeout(greeting, 1000))
+form.addEventListener('submit', handleGreeting)
+
+
+
+// Pseudocode
```

Comment:
Love the pseudo code, a suggestion is to put it in the start of the code. So the user understands the flow better.

Reply:
Totally agree. Good tip!